Keycard - Import a seed phrase into a new Keycard

Parveshdhull opened the pull request:
fixes https://github.com/status-im/status-mobile/issues/21932

### Video
https://github.com/user-attachments/assets/de65611e-b1ca-4cdc-a3a5-6709c9b394d6


status: ready

### Found Issues 

1. [Flow UI issues](https://github.com/status-im/status-mobile/pull/21947#issuecomment-2602647345)  
   - [ ] Fixed  (checked by dev)  
   - [ ] Verified (checked by QA)

2. [The app gets stuck while scanning a different Keycard](https://github.com/status-im/status-mobile/pull/21947#issuecomment-2602801840)  
   - [ ] Fixed  (checked by dev)  
   - [ ] Verified (checked by QA)  

3. [ISSUE 3: Not empty keycard can be used after PIN during Create Profile flow
](https://github.com/status-im/status-mobile/pull/21947#issuecomment-2604383089)  
   - [ ] Fixed  (checked by dev)  
   - [ ] Verified (checked by QA)

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (31)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 6b7a5002 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21947/1/) | 2025-01-17 09:29:37 | ~5 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250117-092421-6b7a50-pr21947-tests.log) |
| :heavy_check_mark: | 6b7a5002 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21947/1/) | 2025-01-17 09:33:46 | ~9 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250117-092421-6b7a50-pr21947-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250117-092421-6b7a50-pr21947-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 6b7a5002 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21947/1/) | 2025-01-17 09:34:33 | ~10 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250117-092421-6b7a50-pr21947-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250117-092421-6b7a50-pr21947-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 6b7a5002 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21947/1/) | 2025-01-17 09:36:40 | ~12 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/QwJUv9) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2FQwJUv9&size=400x400&errorCorrectionLevel=L)|
| | | | | | | |
| :x: | d16f13cb | [#2](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21947/2/) | 2025-01-19 09:26:20 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21947/2/consoleText) |
| :heavy_check_mark: | d16f13cb | [#2](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21947/2/) | 2025-01-19 09:31:28 | ~8 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250119-092326-d16f13-pr21947-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250119-092326-d16f13-pr21947-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | d16f13cb | [#2](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21947/2/) | 2025-01-19 09:31:58 | ~8 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250119-092326-d16f13-pr21947-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250119-092326-d16f13-pr21947-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | d16f13cb | [#2](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21947/2/) | 2025-01-19 09:33:54 | ~10 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/1dyBjZ) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2F1dyBjZ&size=400x400&errorCorrectionLevel=L)|
| | | | | | | |
| :heavy_check_mark: | 83260f49 | [#3](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21947/3/) | 2025-01-19 11:37:16 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250119-113223-83260f-pr21947-tests.log) |
| :heavy_check_mark: | 83260f49 | [#3](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21947/3/) | 2025-01-19 11:41:22 | ~9 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250119-113218-83260f-pr21947-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250119-113218-83260f-pr21947-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 83260f49 | [#3](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21947/3/) | 2025-01-19 11:42:00 | ~9 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/Bt8nM4) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2FBt8nM4&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 83260f49 | [#3](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21947/3/) | 2025-01-19 11:42:07 | ~9 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250119-113223-83260f-pr21947-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250119-113223-83260f-pr21947-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| | | | | | | |
| :heavy_check_mark: | 361645b2 | [#4](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21947/4/) | 2025-01-19 11:48:49 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250119-114357-361645-pr21947-tests.log) |
| :heavy_check_mark: | 361645b2 | [#4](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21947/4/) | 2025-01-19 11:53:17 | ~9 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250119-114353-361645-pr21947-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250119-114353-361645-pr21947-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 361645b2 | [#4](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21947/4/) | 2025-01-19 11:53:36 | ~9 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/s3v15S) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2Fs3v15S&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 361645b2 | [#4](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21947/4/) | 2025-01-19 11:53:44 | ~9 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250119-114357-361645-pr21947-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250119-114357-361645-pr21947-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| | | | | | | |
| :heavy_check_mark: | e49c5993 | [#5](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21947/5/) | 2025-01-19 13:01:05 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250119-125644-e49c59-pr21947-tests.log) |
| :heavy_check_mark: | e49c5993 | [#5](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21947/5/) | 2025-01-19 13:05:15 | ~8 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250119-125644-e49c59-pr21947-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250119-125644-e49c59-pr21947-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | e49c5993 | [#5](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21947/5/) | 2025-01-19 13:05:43 | ~8 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250119-125644-e49c59-pr21947-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250119-125644-e49c59-pr21947-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | e49c5993 | [#5](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21947/5/) | 2025-01-19 13:09:33 | ~12 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/FXs3TT) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2FFXs3TT&size=400x400&errorCorrectionLevel=L)|
| | | | | | | |
| :x: | 79174f75 | [#6](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21947/6/) | 2025-01-20 09:20:10 | ~33 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21947/6/consoleText) |
| :x: | 79174f75 | [#7](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21947/7/) | 2025-01-20 09:22:43 | ~15 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21947/7/consoleText) |
| | | | | | | |
| :x: | 7d100bfa | [#8](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21947/8/) | 2025-01-20 09:23:46 | ~14 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21947/8/consoleText) |
| :heavy_check_mark: | 7d100bfa | [#7](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21947/7/) | 2025-01-20 09:27:38 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250120-092331-7d100b-pr21947-tests.log) |
| :heavy_check_mark: | 7d100bfa | [#7](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21947/7/) | 2025-01-20 09:30:39 | ~7 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250120-092331-7d100b-pr21947-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250120-092331-7d100b-pr21947-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 7d100bfa | [#7](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21947/7/) | 2025-01-20 09:31:16 | ~7 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250120-092331-7d100b-pr21947-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250120-092331-7d100b-pr21947-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 7d100bfa | [#9](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21947/9/) | 2025-01-20 10:24:23 | ~15 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/jR7Vw8) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2FjR7Vw8&size=400x400&errorCorrectionLevel=L)|
| | | | | | | |
| :heavy_check_mark: | f51f924b | [#8](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21947/8/) | 2025-01-20 12:03:30 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250120-115837-f51f92-pr21947-tests.log) |
| :heavy_check_mark: | f51f924b | [#8](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21947/8/) | 2025-01-20 12:07:48 | ~9 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250120-115833-f51f92-pr21947-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250120-115833-f51f92-pr21947-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | f51f924b | [#8](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21947/8/) | 2025-01-20 12:08:20 | ~9 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250120-115833-f51f92-pr21947-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250120-115833-f51f92-pr21947-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | f51f924b | [#10](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21947/10/) | 2025-01-20 12:09:44 | ~11 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/FQ9yzB) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2FFQ9yzB&size=400x400&errorCorrectionLevel=L)|
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 8ce25616 | [#9](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21947/9/) | 2025-01-22 10:16:04 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250122-101133-8ce256-pr21947-tests.log) |
| :heavy_check_mark: | 8ce25616 | [#9](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21947/9/) | 2025-01-22 10:19:25 | ~7 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250122-101129-8ce256-pr21947-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250122-101129-8ce256-pr21947-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 8ce25616 | [#9](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21947/9/) | 2025-01-22 10:21:15 | ~9 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250122-101129-8ce256-pr21947-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250122-101129-8ce256-pr21947-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 8ce25616 | [#11](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21947/11/) | 2025-01-22 10:22:52 | ~11 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/9xdYfH) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2F9xdYfH&size=400x400&errorCorrectionLevel=L)|
| | | | | | | |
| :heavy_check_mark: | 65186461 | [#10](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21947/10/) | 2025-01-23 06:43:08 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250123-063812-651864-pr21947-tests.log) |
| :heavy_check_mark: | 65186461 | [#10](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21947/10/) | 2025-01-23 06:47:09 | ~9 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250123-063808-651864-pr21947-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250123-063808-651864-pr21947-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 65186461 | [#10](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21947/10/) | 2025-01-23 06:47:51 | ~9 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250123-063812-651864-pr21947-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250123-063812-651864-pr21947-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 65186461 | [#12](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21947/12/) | 2025-01-23 06:48:32 | ~10 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/QVUH6f) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2FQVUH6f&size=400x400&errorCorrectionLevel=L)|

Parveshdhull commented:
## Summary

The PR implements a flow for importing a seed phrase into a new keycard and creating an account using that seed phrase. It is essentially a combination of the "recovery phrase flow" and the "Create profile using an empty keycard" flow. Some important points:-

1. In the "Enter recovery phrase" flow in [Figma](https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1/Keycard?node-id=5045-49282&m=dev), there is a screen displaying the error message: "The recovery phrase you entered is linked to a key pair already added to this device," along with a button labeled "Enter another recovery phrase." However, in our standard recovery flow, when `:onboarding/seed-phrase-validated` is dispatched, we show a popup(as shows in above video) that allows the user to open the profiles screen and access the already existing profile (profile selection functionality is currently broken, and I’ve added a comment in the code). Given this, I believe both flows should use the same logic, and we need to decide which approach we want to keep.

Unrelated, but it might be worth considering treating reusable flows as components and defining them in a single place to avoid discrepancies. We could then simply add a label when we need to duplicate a flow, similar to how we handle the "Try Again" button flows.

2. Once the seed phrase is entered, we are using the keycard base flow, and since the base flow doesn’t include a biometric screen, this flow doesn’t either. By the way, the position of the biometric screen is not consistent across our app, which adds complexity to development. I think we should move the biometric step to show after account creation, as I also mentioned [here](https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1?node-id=3-89231#1093892392).

3. "Importing a seed phrase into a new keycard" flow includes an additional [intermediary screen](https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1/Keycard?node-id=5045-51884&t=mk1pIjmK0OVVdGiz-0). Do we need it? I’m also unsure when we should display this screen, as mentioned in this [issue](https://github.com/status-im/status-mobile/issues/21673), where even the visibility of the "Preparing Status for you" screen is sometimes too short.

I don’t think any of these points are blockers for the PR, but please let me know if you think otherwise. @flexsurfer @ilmotta

Parveshdhull commented on src/status_im/contexts/onboarding/events.cljs:
```diff
@@ -115,14 +128,7 @@
-               :on-accept           (fn []
-                                      (re-frame/dispatch [:pop-to-root :screen/profile.profiles])
-                                      (re-frame/dispatch [:profile/profile-selected key-uid]))
-               :on-cancel           #(re-frame/dispatch [:pop-to-root :multiaccounts])}]]}
```
We don't have a :multiaccounts root. Interestingly, pop-to-root didn't throw any errors, even with the incorrect root name, and it just worked because we also call dismiss-all-modals when invoking it.

flexsurfer commented on src/status_im/navigation/screens.cljs:
```diff
@@ -966,7 +966,13 @@
-    :component keycard.create/ready-to-add}])
+    :component keycard.create/ready-to-add}
+
+   {:name      :screen/keycard.enter-seed-phrase
```
could you elaborate please on why new screen id is needed?

Parveshdhull replied:
Hi @flexsurfer, Thank you very much for reviewing the PR.

I initially tried using `:screen/onboarding.enter-seed-phrase` for this flow, but that screen id has some hardcoded elements like `:layout options/onboarding-transparent-layout`, `:animations transitions/push-animations-for-transparent-background`, and is tailored specifically to the onboarding flow (e.g., with use of dynamic overlay and navigate-to-within-stack).
It didn’t work well for the keycard flow, as you can see in the "before" vs. "after" [video](https://github.com/user-attachments/assets/c26af552-3f71-482f-a442-9b77701c9d03)

flexsurfer replied:
@Parveshdhull yeah onboarding is different, that why we have 
```clojure
{:name      :screen/use-recovery-phrase-dark
 :metrics   {:track? true}
 :options   {:theme                  :dark
             :modalPresentationStyle :fullScreen}
 :component enter-seed-phrase/view}
```

which looks identical to your

Parveshdhull replied:
Removed new screen-id, Thank you very much for sharing this.

status-im-auto commented:
## 88% of end-end tests have passed
```
Total executed tests: 8
Failed tests: 1
Expected to fail tests: 0
Passed tests: 7
```
```
IDs of failed tests: 702843 
```
<h3>Failed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=feat/keycard-seed-phrase-flow&PR_ID=21947&APK_NAME=21947.apk&TR_CASE_IDS=702843">Rerun failed tests</a></li><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3768563">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><p><blockquote><div>Device 2: Tap on found: Button </div><div>Device 2: Attempt 0 is successful clicking close-activity-center </div></blockquote></p>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:318: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:388: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:167: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:164: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:154: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:138: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'shoFcUTpGwfFF7WpjPmw')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESXQJEdlgTeFd64t8f">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESg8PbQLOkYa5iI5tp">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (7)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3768561">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESpF8y98JGhT4wVtNz">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3768564">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESpF8y98JGhT4wVtNz">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3768567">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3768568">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3768562">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESY65DCYIP30x4xJ0z">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESI7G16XRhndcztoF2">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3768566">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3768565">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table></details>

mariia-skrypnyk commented:
Hi @Parveshdhull !

Thanks for such an important logic implementation!

Here are 2 UI issues, by now I will group them into one :

## Issue 1: Flow UI issues

- White dots instead of green ones as designed on Import flow and Create Profile flow
*we have a lot of Keycard flows using them, let me know if it's complicated to fix in scope of this PR

**Design vs Implementation**

![Screenshot 2025-01-20 at 15 55 30](https://github.com/user-attachments/assets/d47c2b0f-ac37-4bab-b90f-6055b9dbcfd2)

https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1/Keycard?node-id=5045-52037&m=dev

- Not the correct screen after PIN setting

**Design vs Implementation**

![Screenshot 2025-01-20 at 15 46 11](https://github.com/user-attachments/assets/59e33dbd-6ad1-45b6-884b-25e653d4e5fc)

**Steps:**
1) install app
2) create user
3) use empty keycard
4) choose import recovery phrase
5) scan empty keycard
6) set PIN

https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1/Keycard?node-id=5045-51925&m=dev

mariia-skrypnyk commented:
## ISSUE 2: The app gets stuck while scanning a different Keycard

**Steps:**
1) install app
2) choose create new profile
3) scan Keycard 1   
4) go through the flow 
5) scan Keycard 2 (with a different key pair on it)

**Result:** The app gets stuck on Preparing screen

https://github.com/user-attachments/assets/177eca26-d595-43dd-9c78-d19b2c410676

**Expected result:** 

![Screenshot 2025-01-20 at 17 08 09](https://github.com/user-attachments/assets/2612c7fa-95ca-4f51-b135-262081545deb)

https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1/Keycard?node-id=5045-52070&m=dev

ilmotta commented on src/status_im/contexts/keycard/create/events.cljs:
```diff
@@ -106,3 +106,10 @@
+   (if (contains? (:profile/profiles-overview db) key-uid)
+     {:dispatch [:onboarding/multiaccount-already-exists key-uid]}
+     {:db       (assoc-in db [:keycard :create :masked-phrase] masked-seed-phrase)
+      :dispatch [:keycard/create.create-or-enter-pin]})))
```
According to our guidelines, we use `:db` or `:fx` effects only.

ilmotta commented on src/status_im/contexts/onboarding/events.cljs:
```diff
@@ -115,14 +128,7 @@
-                                      (re-frame/dispatch [:pop-to-root :screen/profile.profiles])
-                                      (re-frame/dispatch [:profile/profile-selected key-uid]))
-               :on-cancel           #(re-frame/dispatch [:pop-to-root :multiaccounts])}]]}
+       {:dispatch [:onboarding/multiaccount-already-exists key-uid]}
```
Same as another comment, better if we stay consistent with our guidelines and only use `:db` or `:fx` effects.

ilmotta reviewed (approved):
> In the "Enter recovery phrase" flow in [Figma](https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1/Keycard?node-id=5045-49282&m=dev), there is a screen displaying the error message: "The recovery phrase you entered is linked to a key pair already added to this device," along with a button labeled "Enter another recovery phrase." However, in our standard recovery flow, when :onboarding/seed-phrase-validated is dispatched, we show a popup(as shows in above video) that allows the user to open the profiles screen and access the already existing profile (profile selection functionality is currently broken, and I’ve added a comment in the code). Given this, I believe both flows should use the same logic, and we need to decide which approach we want to keep.

@Parveshdhull, I don't have any strong preference regarding this point, but I find the current solution showing a pop-up slightly worse than what's in Figma because the user loses context by being redirected to the profile selection. The solution in Figma is consistent with other error messages that can occur in the recovery phrase screen, so that's mainly why I prefer it. If the effort is small seems worth improving to match the UX in Figma. Not a blocker to this PR IMO.

> Unrelated, but it might be worth considering treating reusable flows as components and defining them in a single place to avoid discrepancies. We could then simply add a label when we need to duplicate a flow, similar to how we handle the "Try Again" button flows.

Good idea @Parveshdhull.

> I think we should move the biometric step to show after account creation, as I also mentioned [here](https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1?node-id=3-89231#1093892392).

Agreed 👍🏼 This inconsistency is creating unnecessary complexity.

------

> "Importing a seed phrase into a new keycard" flow includes an additional [intermediary screen](https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1/Keycard?node-id=5045-51884&t=mk1pIjmK0OVVdGiz-0). Do we need it? I’m also unsure when we should display this screen, as mentioned in this https://github.com/status-im/status-mobile/issues/21673, where even the visibility of the "Preparing Status for you" screen is sometimes too short.

Just confirming if I understood your suggestion, per screenshots below, the user would go from the first screen to the second directly, possibly waiting a few seconds? If that's the case, I think it should be fine as long as it doesn't take too long on average (from my usage it's always been relatively fast). Nice suggestion to simplify.

<img src="https://github.com/user-attachments/assets/7a39162e-07f6-4c0a-b3bb-0298dcb29bad" width="250" />

<img src="https://github.com/user-attachments/assets/808fcd27-e3fb-42dd-850d-a71e4db170a4" width="250" />

mariia-skrypnyk commented:
## ISSUE 3: Not empty keycard can be used after PIN during Create Profile flow

**Steps:**

1) Install app
2) Create Profile using empty Keycard
3) Logout and go to Create Profile one more time
4) Choose another empty Keycard in the beginning of Create Profile with empty Keycard flow
5) Go through the flow till PIN set/check
6) Scan Keycard from step 2 (with a key pair on it)

**Result:** 
Keycard will be valid and key pair will be re-written with a new one

**Expected result:**

This is a different Keycard screen should be shown
![Screenshot 2025-01-21 at 11 51 50](https://github.com/user-attachments/assets/e4faa217-34f2-4eae-8964-c67a576bff0b)

https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1/Keycard?node-id=3-89231&p=f&m=dev

Parveshdhull committed:
Keycard - Import a seed phrase into a new Keycard

Parveshdhull commented:
Thank you @ilmotta for reviewing the PR. Addressed required changes.

> If the effort is small seems worth improving to match the UX in Figma.

Thank you, added task in https://github.com/status-im/status-mobile/issues/21970

> Just confirming if I understood your suggestion, per screenshots below, the user would go from the first screen to the second directly, possibly waiting a few seconds? 

Yes

> If that's the case, I think it should be fine as long as it doesn't take too long on average (from my usage it's always been relatively fast). Nice suggestion to simplify

Thank you

Parveshdhull commented:
Thanks, @mariia-skrypnyk, for testing the PR.

The "Import a seed phrase into a new Keycard" flow combines the existing seed phrase and Keycard base flows, so this PR inherits their limitations and issues.

The reported issues are important but addressing them all at once would add complexity. I’ve created a separate issue for these bugs (including the info button update task from the old [issue](https://github.com/status-im/status-mobile/issues/21675)).

Since these issues are already part of the develop branch and were not introduced by this PR, we can focus on testing only the new functionality of the PR. Please let me know if this is not the case for any of the issues.

mariia-skrypnyk commented:
Hi @Parveshdhull !

Understand your concern!
Not sure I agree with Issue 3: the problem is that we have no check that the keycard already has a key pair on it and we rewrite it.

Does it make sence? Or you still think it is out of the current PR's scope?

UPD: I found issue in repo that we do not have this logic for now https://github.com/status-im/status-mobile/issues/21647 and probably that's why it was not included by you (nothing to include).

So, Issue 3 with https://github.com/status-im/status-mobile/issues/21647 should also exist as a separate tasks.

Failed e2e is not related.
PR can be merged.
